Inicio do modo de contingência

ImplantiSolucoes opened the pull request:
Acabei bagunçando o antigo fork e fiz um novo e passei as alterações corrigidas...

maiconkkl committed:
Corrigindo a geração do QR Code

Foi corrigido a geração do QRCode para usar os paramentros corretos e considerar o tipo_emissao, mantendo a formar antiga quando 1 e usando a nova forma quando for diferente.

maiconkkl committed:
Merge branch '11.0' of https://github.com/Trust-Code/odoo-brasil into 11.0

maiconkkl committed:
Criando paramentros fixo para contingência

Foi criado parametros fixo para contingência, sera melhor definidos no futuro.

maiconkkl committed:
Correções

Foi feitas algumas correções que o danimar pediu no PR #846

maiconkkl committed:
.

maiconkkl committed:
Alterado pra usar tradução

Criado o arquivo de tradução para parar de gerar erro.

maiconkkl committed:
.

.

maiconkkl committed:
.

.

danimaribeiro commented on br_nfe/models/invoice_eletronic.py:
```diff
@@ -373,16 +382,18 @@ def _prepare_eletronic_invoice_item(self, item, invoice):
         }
         if item.tipo_produto == 'service':
-            retencoes = item.pis_valor_retencao + \
-                item.cofins_valor_retencao + item.inss_valor_retencao + \
```
Evita modificar coisas desnecessárias pois me ajuda depois ao migrar as mudanças pra versão 12, evita conflitos de merge.
E também não introduz bugs desnecessários.

ImplantiSolucoes replied:
ah... só dei uma melhorada no visual, nada de mais... não mudei nada que vai prejudicar

ImplantiSolucoes replied:
São mais para não gerar advertências no pycharm mesmo, pois aquilo me dar tick nervoso e outra que é padrões de código mesmo

ImplantiSolucoes replied:
Só ficou confuso a seguinte situação: Quando uma nota for transmitida, quando ela vai ativar o modo de contingencia ou não ?

Na minha opinião poderia ser o seguinte... 
1. Validar
    * Faz validações padrões ja existente
    * Gera o XML
    * Passa ele no esquema de validação
2. Transmissão
    * Testa para ver se o servidor da sefaz esta funcionando (Consulta status)
        - Gera o xml e Tenta transmitir caso esteja tudo certo
        - Gera o xml com os parâmetros de contingência
        - Gera pdf de modelo de contingencia para impressão
        - Coloca em uma fila especial para fica verificando o servidor da sefaz esta ativo, quando estive tudo certo, faz a transmissão

ImplantiSolucoes commented:
@danimaribeiro só ficou confuso a seguinte situação: Quando uma nota for transmitida, quando ela vai ativar o modo de contingencia ou não ?

Na minha opinião poderia ser o seguinte... 
1. Validar
    * Faz validações padrões ja existente
    * Gera o XML
    * Passa ele no esquema de validação
2. Transmissão
    * Testa para ver se o servidor da sefaz esta funcionando (Consulta status)
        - Gera o xml e Tenta transmitir caso esteja tudo certo
        - Gera o xml com os parâmetros de contingência
        - Gera pdf de modelo de contingencia para impressão
        - Coloca em uma fila especial para fica verificando o servidor da sefaz esta ativo, quando estive tudo certo, faz a transmissão

maiconkkl committed:
desfazendo alteração

desfazendo alteração desnecessaria

nogueira-raphael commented:
Gravar o Digest Value no banco? Qual a necessidade?

ImplantiSolucoes commented:
> Gravar o Digest Value no banco? Qual a necessidade?

É necessário porque o xml pode ser excluído ou modificado gerando um certo probleminha com os dados

nogueira-raphael commented:
@ImplantiSolucoes como a emissão em contingência depende do XML assinado, acho que a responsabilidade de gerar o qrcode em contingência deveria ficar na pytrustnfe, eu fiz assim...
Como no core do odoo-brasil foi implementado a geração do qrcode no odoo, estou usando muda um pouco, mas posso ver se da pra mover, mas essa responsabilidade na minha opinião seria melhor na pytrustnfe.

ImplantiSolucoes commented:
> @ImplantiSolucoes como a emissão em contingência depende do XML assinado, acho que a responsabilidade de gerar o qrcode em contingência deveria ficar na pytrustnfe, eu fiz assim...
> Como no core do odoo-brasil foi implementado a geração do qrcode no odoo, estou usando muda um pouco, mas posso ver se da pra mover, mas essa responsabilidade na minha opinião seria melhor na pytrustnfe.

Curti a ideia... do mesmo jeito que pega o Digest Value poderia pega o QRCode, logo isso reduz um campo no banco de dados que é a ideia ideal do @danimaribeiro... Só depende ver se o @danimaribeiro  aceita passa pra o pytrustnfe

danimaribeiro commented:
> @danimaribeiro só ficou confuso a seguinte situação: Quando uma nota for transmitida, quando ela vai ativar o modo de contingencia ou não ?
> 
> Na minha opinião poderia ser o seguinte...
> 
> 1. Validar
>    
>    * Faz validações padrões ja existente
>    * Gera o XML
>    * Passa ele no esquema de validação
> 2. Transmissão
>    
>    * Testa para ver se o servidor da sefaz esta funcionando (Consulta status)
>      
>      * Gera o xml e Tenta transmitir caso esteja tudo certo
>      * Gera o xml com os parâmetros de contingência
>      * Gera pdf de modelo de contingencia para impressão
>      * Coloca em uma fila especial para fica verificando o servidor da sefaz esta ativo, quando estive tudo certo, faz a transmissão

- No documento eletrônico deve haver um status novo "contigencia".
- Fila específica para enviar nesse status - ok
- Tenta-se o envio diretamente, sem consulta status, em caso de erro se emite em contingência.
- É bom ter um parâmetro da empresa que diz se passa para contingência automaticamente.
- Sim, pode ser passado ao pytrustnfe a geração da url do QRCode, só evitar mudar os métodos (parâmetros e retorno de método).

maiconkkl committed:
Ultilizando a função de gerar qr

Ultilizando a função de gerar qr pelo pytrustnfe;
implementado no status de contigencia.

ImplantiSolucoes committed:
Merge branch '11.0' into 11.0

ImplantiSolucoes committed:
Merge branch '11.0' into 11.0

ImplantiSolucoes committed:
Merge branch '11.0' into 11.0

ImplantiSolucoes committed:
Merge branch '11.0' into 11.0

maiconkkl committed:
Ajuste na tabela de NCM

ImplantiSolucoes committed:
Merge branch '11.0' into 11.0

ImplantiSolucoes committed:
Merge branch '11.0' into 11.0

maiconkkl committed:
Merge branch '11.0' of https://github.com/Trust-Code/odoo-brasil into 11.0

ImplantiSolucoes commented:
Corrigido conflitos de alteração.

ImplantiSolucoes committed:
Merge branch '11.0' into 11.0

ImplantiSolucoes committed:
Merge branch '11.0' into 11.0

maiconkkl committed:
Chamando função para carregar o QRCode

maiconkkl committed:
Correção tipo de dados

Correção do tipo de dados

ImplantiSolucoes committed:
Merge branch '11.0' into 11.0

ImplantiSolucoes committed:
Merge branch '11.0' into 11.0
